File: infrastructure/src/main/java/com/study/platform/global/redis/RedisMessagePublisherAdapter.java
package com.study.platform.global.redis;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class RedisMessagePublisherAdapter implements RedisMessagePublisher {

    private final StringRedisTemplate stringRedisTemplate;

    @Override
    public void publish(String channel, String message) {
        stringRedisTemplate.convertAndSend(channel, message);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/global/redis/RedisMessagePublisherAdapter.kt
package com.study.platform.global.redis

import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.stereotype.Component

@Component
class RedisMessagePublisherAdapter(
    private val stringRedisTemplate: StringRedisTemplate
) : RedisMessagePublisher {

    override fun publish(channel: String, message: String) {
        stringRedisTemplate.convertAndSend(channel, message)
    }
}
